### DISCUSSION TITLE
program-service cert expire problem.

### DISCUSSION THREAD

**Discussion by heungheung** [Sunbird-coKreat/Community#74] [Bugs]
Program Service is build with pretty old nodejs docker image
https://github.com/Sunbird-coKreat/program-service/blob/release-5.1.0_RC1/Dockerfile
ca cert store in the image got ca cert already expired

axios is making post REST call to the FQDN of the base_url and will reject the TLS server cert.

<img width="1910" alt="Screenshot 2023-04-15 at 5 06 44 PM" src="https://user-images.githubusercontent.com/23467024/232203087-1bece530-0050-49cd-91b6-caa1eba6788f.png">

This problem affects all nodejs microservices which
- use node8 as the base image
- make https API call to FQDN of the target endpoint (instead of taking to the nginx private ingress or k8s service directly)

of corresponding build blocks, e.g. knowledgemw in ED. 

X-post similar problem here in ED discussions
https://github.com/orgs/Sunbird-Ed/discussions/457

IMHO, we better to move all nodejs app to a newer version.

**Comment by Krishnaj20** [comment]
thanks @heungheung for sharing this information.

@mmanimegalai - please get this feedback added to the backlog.
